### PR TITLE
[Profiling] add profiling code to get kernel submit/execute time

### PR DESCRIPTION
**Pull request by Dewei-Wang-sh**
take test/PlaidML/OpTest.EltwiseAdd.mlir for example, we can get the following result.
the kernel execution time is 0.082560 ms
the kernel submission time is 97647.601562 ms  
( this duration is obviously wrong cause the test is done within 3 seconds, still trying to find the reason)

Please review these guidelines to help with the review process:
- [x] Have you provided a meaningful PR description?
- [ ] Have you added a test, a reproducer, or a reference to an issue with a reproducer?
- [x] Have you tested your changes locally for CPU and GPU devices?
- [x] Have you made sure that new changes do not introduce compiler warnings?
- [x] If this PR is a work in progress, are you filing the PR as a draft?
- [x] Have you organized your commits logically and ensured each can be built by itself?

### CODE REVIEW

File: include/imex/Transforms/Passes.td [hunk 1]
```diff
@@ -46,6 +46,7 @@ def InsertGPUAllocs : Pass<"insert-gpu-allocs", "::mlir::func::FuncOp"> {
 def SetSPIRVCapabilities : Pass<"set-spirv-capabilities"> {
   let summary = "Sets Spirv capabilities";
   let constructor = "imex::createSetSPIRVCapabilitiesPass()";
+  let dependentDialects = ["::mlir::spirv::SPIRVDialect"];
```
Comment (Dewei-Wang-sh) [inline include/imex/Transforms/Passes.td:49]: without this, it will break when run this pass alone.

**Review by silee2 — APPROVED** [review]
LGTM

**Comment by nbpatel** [comment]
Thanks for this PR.
Can we add something similar for L0 as well?

**Review by chencha3 — APPROVED** [review]
LGTM

**Pull request merged**